apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestPlugLatencyActivity.java

```diff
@@ -0,0 +1,214 @@
+    MyAudioDeviceCallback mDeviceCallback = new MyAudioDeviceCallback();
+    private AudioManager mAudioManager;
+
+    private volatile int mPlugCount = 0;
+
+    class MyAudioDeviceCallback extends AudioDeviceCallback {
+        private HashMap<Integer, AudioDeviceInfo> mDevices
+                = new HashMap<Integer, AudioDeviceInfo>();
+
+        @Override
+        public void onAudioDevicesAdded(AudioDeviceInfo[] addedDevices) {
+            for (AudioDeviceInfo info : addedDevices) {
+                mDevices.put(info.getId(), info);
+                log("Device Added");
+                log(adiToString(info));
+            }
+
+            updateLatency();
+        }
+
+        public void onAudioDevicesRemoved(AudioDeviceInfo[] removedDevices) {
+            for (AudioDeviceInfo info : removedDevices) {
+                mDevices.remove(info.getId());
+                log("Device Removed");
+                log(adiToString(info));
+            }
+
+            updateLatency();
+        }
+    }
+
+    @Override
+    protected void inflateActivity() {
+        setContentView(R.layout.activity_test_plug_latency);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mInstructionsTextView = (TextView) findViewById(R.id.text_instructions);
+        mPlugTextView = (TextView) findViewById(R.id.text_plug_events);
+        mAutoTextView = (TextView) findViewById(R.id.text_log_device_report);
+
+        mAudioManager = (AudioManager) getSystemService(Context.AUDIO_SERVICE);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        addAudioDeviceCallback();
+    }
+
+    @Override
+    protected void onStop() {
+        removeAudioDeviceCallback();
+        super.onStop();
+    }
+
+    @TargetApi(23)
+    private void addAudioDeviceCallback(){
+        // Note that we will immediately receive a call to onDevicesAdded with the list of
+        // devices which are currently connected.
+        mAudioManager.registerAudioDeviceCallback(mDeviceCallback, null);
+    }
+
+    @TargetApi(23)
+    private void removeAudioDeviceCallback(){
+        mAudioManager.unregisterAudioDeviceCallback(mDeviceCallback);
+    }
+
+    @Override
+    public String getTestName() {
+        return "Plug Latency";
+    }
+
+    int getActivityType() {
+        return ACTIVITY_TEST_DISCONNECT;
+    }
+
+    @Override
+    boolean isOutput() {
+        return true;
+    }
+
+    @Override
+    public void setupEffects(int sessionId) {
+    }
+
+    // Write to status and command view
+    private void setInstructionsText(final String text) {
+        runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                mInstructionsTextView.setText(text);
+            }
+        });
+    }
+
+    public void startAudioTest() throws IOException {
+        startAudio();
+    }
+
+    private long getLatencyMs() {
```

Comment:
This does a lot more than a normal "get" method. Maybe "measureLatencyMs".

Reply:
Good idea!